### PR TITLE
prepare modal for template representation

### PR DESCRIPTION
**Pull request by Valexr**

### CODE REVIEW

File: src/services/api.ts [hunk 1]
```diff
@@ -39,22 +37,22 @@ export async function delDataSource({ id, query }: { id: number, query?: string
...
 
-export async function getCalculationsEngines(): Promise<void> {
+export async function getCalculationEngines(): Promise<string[]> {
 	return getJSON(`/calculations/engines`);
 }
 
-export async function setCalculation({ dataId, engine = 'dummy', input, workflow = 'workflow' }:
+export async function getCalculationEngine(engine = 'dummy'): Promise<EngineDTO> {
+	return fetchJSON(
+		`${API_HOST}/calculations/template?engine=${engine}`,
```
Comment (blokhin) [inline src/services/api.ts:50]: @Valexr why do we need `API_HOST` here? It can work via BFF as well.

File: src/components/Editor/Form.svelte [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+</script>
+
+<script lang="ts">
+	export let template = '';
+	export let schema: { [key: string]: any } = {};
+	export let input = '';
+
+	const { description, properties } = schema;
+
+	console.log(template, schema, input);
```
Comment (blokhin) [inline src/components/Editor/Form.svelte:39]: @Valexr next time please do not forget to comment it out!

File: src/components/Editor/jsontree/jsontree.ts [hunk 3]
```diff
@@ -0,0 +1,303 @@
...
+export function collapse(node) {
+	traverse(node, function (child) {
+		child.isExpanded = false;
+		if (child.depth > node.depth) child.el.classList.add(classes.HIDDEN);
+		setCaretIconRight(child);
+	});
+}
+
+export function filter(node, keys) {
+	console.log(keys);
```
Comment (blokhin) [inline src/components/Editor/jsontree/jsontree.ts:265]: Again non-commented.

File: src/stores/calculations.ts [hunk 4]
```diff
@@ -18,8 +18,8 @@ type CalculationDTO = {
 
 let unsubscribe: Unsubscriber
```
Comment (blokhin) [inline src/stores/calculations.ts:19]: Why semicolon at the end of string is not used?

File: src/services/api.ts (hunk 1 above)
Comment (Valexr) [reply src/services/api.ts:50]: ok, will try it...

File: src/components/Editor/Form.svelte (hunk 2 above)
Comment (Valexr) [reply src/components/Editor/Form.svelte:39]: it's very drafted code...

File: src/components/Editor/jsontree/jsontree.ts (hunk 3 above)
Comment (Valexr) [reply src/components/Editor/jsontree/jsontree.ts:265]: again draft 🤓

File: src/stores/calculations.ts (hunk 4 above)
Comment (Valexr) [reply src/stores/calculations.ts:19]: Maybe VScode missing auto-format in TS...🤔

File: src/services/api.ts (hunk 1 above)
Comment (Valexr) [reply src/services/api.ts:50]: Removes version from url path to bff

**Comment by blokhin** [comment]
Closed in favor of #84

**Pull request closed without merging**